### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by andrewkho**
Updates readme with torchdata plans

see #1196 

### Changes
* Updates README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,49 +1,39 @@
...
-These reasons inspired the creation of `DataPipe` and `DataLoader2`, with a goal to make data loading components more
-flexible and reusable.
+[**What is TorchData?**](#what-is-torchdata) | [**Stateful DataLoader**](#stateful-dataloader) |
+[**Install guide**](#installation) | [**Contributing**](#contributing) | [**License**](#license)
+
+**:warning: June 2024 Status Update: Removing DataPipes and DataLoader V2**
+
+**We are re-focusing the torchdata repo to be an iterative enhancement of torch.utils.data.DataLoader. We do not plan on
+continuing development or maintaining the [`DataPipes`] and [`DataLoaderV2`] solutions, and they will be removed from
+the torchdata repo. We'll also be revisiting the `DataPipes` references in pytorch/pytorch. In release 0.8.0 (July 2024)
```
Comment (gokulavasan) [inline README.md:10]: nit: In release 0.8.0 of torchdata

**Review by gokulavasan — APPROVED** [review]
nit comment about release package name, rest lgtm!

**Comment by andrewkho** [comment]
Ignoring the windows s3 file lister failures, tests are already disabled on main

File: README.md (hunk 1 above)
Comment (andrewkho) [reply README.md:10]: oh crap missed this before landing

**Pull request merged**